traitsui/qt4/list_str_editor.py

```diff
@@ -388,8 +388,8 @@ def _on_activate(self, mi):
         self.activated_index = index = mi.row()
         self.activated = self.adapter.get_item(self.object, self.name, index)
 
-    def _on_context_menu(self, point):
-        """ Handle a context menu request.
+    def _on_mouse_right_click(self, point):
```

Comment:
Umm. I'm not sure why there are two `_on_context_menu` methods defined on `_ListStrEditor`.

If this is the right approach, do we also want to not hook up the non-existent `_on_context_menu` method in `_ListView.__init__`?

Reply:
> do we also want to not hook up the non-existent _on_context_menu method in _ListView.__init__?

I might have misunderstood what you meant... let me try to clarify what is happening here.

The second `_on_context_menu` (now the only `_on_context_menu`) is hooked up in the `_ListView.__init__` already.

This `_on_mouse_right_click` (previously called `_on_context_menu` as well) was as if it was never there. Now it is hooked up with the `_ListView.mouseReleaseEvent`. If we don't hook it up, then we won't be fixing that part of #868 I quoted.

What happened here is that this `_on_mouse_right_click` used to be the only `_on_context_menu`. It was hooked up with the context menu event because right clicking the widget fires that event too (among many other events). Later on we wanted additional control on the context menu behaviour, and added another `_on_context_menu`. And because there was no flake8 telling us the method is defined twice, the first one is effectively removed.